collector/receiver/telemetryapireceiver/receiver.go

```diff
@@ -217,16 +218,31 @@ func (r *telemetryAPIReceiver) createLogs(slice []event) (plog.Logs, error) {
 				}
 				if requestId, ok := record["requestId"].(string); ok {
 					logRecord.Attributes().PutStr(semconv.AttributeFaaSInvocationID, requestId)
+				} else if r.currentFaasInvocationID != "" {
+					logRecord.Attributes().PutStr(semconv.AttributeFaaSInvocationID, r.currentFaasInvocationID)
 				}
 				if line, ok := record["message"].(string); ok {
 					logRecord.Body().SetStr(line)
 				}
 			} else {
+				if r.currentFaasInvocationID != "" {
+					logRecord.Attributes().PutStr(semconv.AttributeFaaSInvocationID, r.currentFaasInvocationID)
+				}
 				// in plain text https://docs.aws.amazon.com/lambda/latest/dg/telemetry-schema-reference.html#telemetry-api-function
 				if line, ok := el.Record.(string); ok {
 					logRecord.Body().SetStr(line)
 				}
 			}
+		} else { // platform events, if subscribed to
+			if el.Type == string(telemetryapi.PlatformStart) {
+				if record, ok := el.Record.(map[string]interface{}); ok {
+					if requestId, ok := record["requestId"].(string); ok {
+						r.currentFaasInvocationID = requestId
```

Comment:
Not as familiar with this code, but any reason you are not logging the request ID after this line?

Reply:
no special reason. I can add a debug level log to this, if it's helpful